logstash-core/lib/logstash/runner.rb

```diff
@@ -424,6 +428,25 @@ def execute
     @log_fd.close if @log_fd
   end # def self.main
 
+  def checkLogConfigurationContainsJavascriptUsage
+     context = LoggerContext.getContext(false)
+     config = context.configuration
+     config_file = config.configuration_source.file
+     # no config file so nothing to check
+     unless config_file
+        return false
+     end
+     logger.info("Log4j configuration path used is: #{config_file.path}")
+     log_config = File.open(config_file.absolute_path).read
+     if log_config =~ /^.*avoid_pipelined_filter\.type\s*=\s*ScriptFilter.*$/ ||
```

Comment:
nit: just checking the script content might be more robust in terms of filter naming, and also potentially avoid matching against a huge string? (assuming the script hasn't been changing)
```ruby
File.open(config_file).any? { |line| line.index('${sys:ls.pipeline.separate_logs} == false') && ! line.strip.start_with?('#') }
```
during an upgrade it might be common for users to comment out old content as they bring in new configuration?
'safest' option would be walking the runtime log4j configuration representation if that doesn't sounds like an overkill.

Reply:
I believe it is the case that the `Script` and `ScriptFilter` is what is not supported, in which case we can narrow our search to just that, regardless of how it is used. This will handle _both_ our own use of these filters/appenders and our user's custom log configurations:

~~~ ruby
      # a line that is not a comment, containing a `type=Script` or `type=ScriptFilter`
      if log_config =~ /^[^#]+\.type\s*=\s*Script(?:Filter)?\s*(?:#|$)/
~~~

Reply:
also fine by me or maybe just checking for an un-commented `/^[^#]+script\.language\s*=\s*JavaScript/i` will do

Reply:
I thought to do the check on runtime log4j configuration but that's an overkill, at the end we want to check if the log4j config properties contains our deprecated script definition.